routing/datasource.go

```diff
@@ -389,12 +394,12 @@ func receiveRouteMatcher(o Options, out chan<- *matcher, quit <-chan struct{}) {
 			for _, err := range errs {
 				o.Log.Error(err)
 			}
-
-			mout = m
+			// TODO: What about the errors? Should we exclude the routes which has errors?
```

Comment:
though, after thinking about it, we should be careful with the slices and maps, because of the memory. Do you know if this current routes slice is thrown away after the update? If not, then there is somewhere a map containing the routes, that could be used. and maybe with a container struct, each of them could be marked depending on whether they errored or succeeded.